ext/standard/streamsfuncs.c

```diff
@@ -756,115 +795,144 @@ static int stream_array_emulate_read_fd_set(zval *stream_array)
 /* }}} */
 
 /* {{{ Runs the select() system call on the sets of streams with a timeout specified by tv_sec and tv_usec */
-PHP_FUNCTION(stream_select)
-{
-	zval *r_array, *w_array, *e_array;
-	struct timeval tv, *tv_p = NULL;
-	fd_set rfds, wfds, efds;
-	php_socket_t max_fd = 0;
-	int retval, sets = 0;
-	zend_long sec, usec = 0;
-	bool secnull;
-	bool usecnull = 1;
-	int set_count, max_set_count = 0;
-
-	ZEND_PARSE_PARAMETERS_START(4, 5)
-		Z_PARAM_ARRAY_EX2(r_array, 1, 1, 0)
-		Z_PARAM_ARRAY_EX2(w_array, 1, 1, 0)
-		Z_PARAM_ARRAY_EX2(e_array, 1, 1, 0)
-		Z_PARAM_LONG_OR_NULL(sec, secnull)
-		Z_PARAM_OPTIONAL
-		Z_PARAM_LONG_OR_NULL(usec, usecnull)
-	ZEND_PARSE_PARAMETERS_END();
+PHP_FUNCTION(stream_select) {
+    zval *r_array, *w_array, *e_array;
+    struct timeval tv, *tv_p = NULL;
+    fd_bigset rfds, wfds, efds;
+    php_socket_t max_fd = 0;
+    int retval, sets = 0;
+    zend_long sec, usec = 0;
+    bool secnull;
+    bool usecnull = 1;
+    int set_count = 0, max_set_count = 0;
+
+    ZEND_PARSE_PARAMETERS_START(4, 5)
+        Z_PARAM_ARRAY_EX2(r_array, 1, 1, 0)
+        Z_PARAM_ARRAY_EX2(w_array, 1, 1, 0)
+        Z_PARAM_ARRAY_EX2(e_array, 1, 1, 0)
+        Z_PARAM_LONG_OR_NULL(sec, secnull)
+        Z_PARAM_OPTIONAL
+        Z_PARAM_LONG_OR_NULL(usec, usecnull)
+    ZEND_PARSE_PARAMETERS_END();
 
-	FD_ZERO(&rfds);
-	FD_ZERO(&wfds);
-	FD_ZERO(&efds);
+#ifdef PHP_WIN32
+	/* Check if the Windows standard handle is redirected to file */
+	long max_fds = FD_SETSIZE;
+#else
+	/* Check if the file descriptor identifier is a terminal */
+	long max_fds = sysconf(_SC_OPEN_MAX);
+#endif
 
-	if (r_array != NULL) {
-		set_count = stream_array_to_fd_set(r_array, &rfds, &max_fd);
+    FD_BIGSET_ZERO(&rfds, max_fds);
+    FD_BIGSET_ZERO(&wfds, max_fds);
+    FD_BIGSET_ZERO(&efds, max_fds);
+
+    if (r_array != NULL) {
+        set_count = stream_array_to_fd_bigset(r_array, &rfds, &max_fd, max_fds);
 		if (set_count > max_set_count)
 			max_set_count = set_count;
-		sets += set_count;
+		sets += set_count;    
 	}
 
-	if (w_array != NULL) {
-		set_count = stream_array_to_fd_set(w_array, &wfds, &max_fd);
+    if (w_array != NULL) {
+        set_count = stream_array_to_fd_bigset(w_array, &wfds, &max_fd, max_fds);
 		if (set_count > max_set_count)
 			max_set_count = set_count;
 		sets += set_count;
 	}
 
-	if (e_array != NULL) {
-		set_count = stream_array_to_fd_set(e_array, &efds, &max_fd);
+    if (e_array != NULL) {
+        set_count = stream_array_to_fd_bigset(e_array, &efds, &max_fd, max_fds);
 		if (set_count > max_set_count)
 			max_set_count = set_count;
-		sets += set_count;
-	}
+        sets += set_count;
```

Comment:
I know this existed before you came along, but can we fix the indentation here so that it doesn't look like it's a part of the if-statement?

Reply:
I think this has been resolved?

Reply:
Looking at in GitHub, I see this:

![image](https://github.com/php/php-src/assets/1883296/b7fbb742-5629-4977-b198-f45e4edd2d49)

which makes it look like `sets += set_count;` while scanning the code, might be part of the `if (set_count > max_set_count)` but it isn't.

Reply:
I think this might be related to the usage of spaces vs. tabs, so once that is fixed, this should look fine.